src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -4,14 +4,112 @@
 Description: MSPileup provides logic behind the pileup WMCore module.
 """
 
+# system modules
+from threading import current_thread
+
 # WMCore modules
+from WMCore.REST.Auth import authz_match
 from WMCore.MicroService.MSCore.MSCore import MSCore
+from WMCore.MicroService.DataStructs.DefaultStructs import OUTPUT_REPORT
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
 
 
 class MSPileup(MSCore):
     """
     MSPileup provides whole logic behind the pileup WMCore module.
     """
 
-    def __init__(self, msConfig, logger=None, **kwargs):
-        super(MSPileup, self).__init__(msConfig, **kwargs)
+    def __init__(self, msConfig, **kwargs):
+        # TODO: so far we are testing this service w/o Rucio access
+        super(MSPileup, self).__init__(msConfig, skipRucio=True)
+        # TODO: if more generic approach will be required we'll need to switch to
+#         super(MSPileup, self).__init__(msConfig, **kwargs)
+
+        # for authz we should provide relevant section in MS configuration
+        # if no role/group is provided then authz_match used in HTTP APIs
+        # will always return valid state. Therefore, for test purposes we can
+        # use empty role and group, while for production setup we should set thenm up
+        authzDefaults = msConfig.get('authz_defaults', {})
+        self.role = authzDefaults.get('role', [])
+        self.group = authzDefaults.get('group', [])
+        self.mgr = MSPileupData(msConfig)
+
+    def status(self):
+        """
+        Provide MSPileup status API. We should extend it to check DB connection, etc.
+
+        :return: status dictionary
+        """
+        summary = dict(OUTPUT_REPORT)
```

Comment:
I do not know since it depends what will be used by metrics. For instance the status may report status of MSPileup instance along with status of underlying data layer stats. Again, I suggest to make necessary changes once we'll address monitoring issue #11429